scripts/o3de/ExportScripts/export_source_ios_xcode.py

```diff
@@ -0,0 +1,122 @@
+#
+# Copyright (c) Contributors to the Open 3D Engine Project.
+# For complete copyright and license terms please see the LICENSE at the root of this distribution.
+#
+# SPDX-License-Identifier: Apache-2.0 OR MIT
+#
+#
+
+import argparse
+import logging
+import sys
+
+import o3de.export_project as exp
+from o3de.export_project import process_command
+import pathlib
+
+def export_ios_Xcode_project(ctx: exp.O3DEScriptExportContext,
+                             tools_build_folder: pathlib.Path,
+                             ios_build_folder: pathlib.Path,
+                             should_build_tools:bool = True,
+                             skip_asset_processing: bool = False,
+                             logger: logging.Logger|None = None):
+    """
+    This function serves as an initial exporter for project to the iOS platform. The steps in this code will generate
+    an Xcode project file containing all necessary build information to produce iOS executables. 
+    In order to build and deploy projects, it is recommended to open the project file in Xcode to configure and build accordingly.
+
+    Note: In order to use this functionality, you must be running this script from a MacOS machine with a valid copy of Xcode
+
+    Instructions to handle iOS projects in Xcode will be provided soon.
+
+    :param ctx:                                     The O3DE Script context provided by the export-command
+    :param tools_build_folder:                      Optional build path to build the tools. (Will default to build/tools if not supplied)
+    :param ios_build_folder:                        The base output path of the generated Xcode Project file for iOS
+    :param should_build_tools:                      Option to build the export process dependent tools (AssetProcessor, AssetBundlerBatch, and dependencies)
+    :param should_build_all_assets:                 Option to process all the assets for the game
+    :param logger:                                  Optional logger to use to log the process and errors
+    """
+    
+    # Use a provided logger or get the current system one
+    if not logger:
+        logger = logging.getLogger()
+        logger.setLevel(logging.ERROR)
+
+    tools_build_folder_str = str(tools_build_folder)
+    ios_build_folder_str = str(ios_build_folder)
+
+    # Optionally build the toolchain needed to process the assets
+    if should_build_tools:
+        process_command(["cmake", "-B", tools_build_folder_str, '-G', "Xcode","-DLY_UNITY_BUILD=ON"], cwd=ctx.project_path)
```

Comment:
The `LY_UNITY_BUILD=ON` argument is not needed.
It is the default value, it is only needed to turn it off.
```suggestion
        process_command(["cmake", "-B", tools_build_folder_str, '-G', "Xcode"], cwd=ctx.project_path)
```